### DISCUSSION TITLE
add text to an image

### DISCUSSION THREAD

**Discussion by bjlockie** [ImageMagick/ImageMagick#4078] [Digital Image Processing]
I'm trying to add text to an image.
I have 3 strings to add but I can't get one to work.
I tried:
$ convert -pointsize 40 -draw 'fill black text 0,0 "When"' "1 1.jpg" "1 1_annotated.jpg"
It creates "1 1_annotated.jpg" but there doesn't seem to be any text.

I was going to add text in each corner but now I'm thinking I might resize the image to add space for the 3 lines.
First thing is getting any text.

**Comment by fmw42** [comment]
Read the image first right after convert. 

`convert "1 1.jpg" -pointsize 40 -draw 'fill black text 0,0 "When"' "1 1_annotated.jpg"`

**Reply by bjlockie** [reply]
I still don't see any text.

**Reply by fmw42** [reply]
try swapping quotes and add your font and gravity setting

`convert "1 1.jpg" -font Arial -pointsize 40 -gravity Northwest -draw "fill black text 0,0 'When'" "1 1_annotated.jpg"`

**Reply by bjlockie** [reply]
Thanks.
I didn't have Ariel but I listed fonts with convert -list fonts and this worked:

 convert "1 1.jpg" -font Courier -pointsize 40 -gravity Northwest -draw "fill white text 0,0 'When'" "1 1_annotated.jpg"

**Reply by fmw42** [reply]
It is spelled Arial not Ariel. But as long as it works for some font, then fine. You can also specify fonts by path to the font file itself.
